project/djangoproject/hw/views.py

```diff
@@ -0,0 +1,48 @@
+from django.views.generic import TemplateView
+from .models import Order
+from django.http import HttpResponse, Http404
+from django.shortcuts import render
+from datetime import datetime, timedelta
+
+
+def client_orders(request: HttpResponse, client_id: int):
+    orders = Order.objects.filter(client__id=client_id)
+    return render(request, 'shopapp/client_orders.html',
+                  context={'orders': orders, 'client_id': client_id})
+
+
+class ClientProductsReport(TemplateView):
+    template_name = 'shopapp/client_product_report.html'
+    allowed_range_names = {
+        'day': 1,
+        'week': 7,
+        'month': 30,
+        'year': 365
+    }
+
+    def get_products(self, client_id: int, days: int):
+        products_info = {}
+        orders_list = []
+        orders = Order.objects.filter(client__id=client_id, register_date=datetime.now()-timedelta(days=days))
```

Comment:
Так у вас же вернуться заказы все с конкретной датой. Но заказы могут быть созданы в разные дни.
```suggestion
        orders = Order.objects.filter(client__id=client_id, register_date__gte=datetime.now()-timedelta(days=days))
```